appcelerator/titanium_mobile_windows
# [TIMOB-24097] Native event handling

- [TIMOB-24097](https://jira.appcelerator.org/browse/TIMOB-24097) Native event handling
- [TIMOB-24099](https://jira.appcelerator.org/browse/TIMOB-24099) Mix WinRT APIs with Titanium APIs

Native event handling for hyperloop using `addEventListener`. 
This also integrates mix with native views with Titanium views, such as adding `Windows.UI.Xaml.Controls.Button` into `Ti.UI.Window`.

```javascript
require('hyperloop').debug = true;

// 
// Currently requiring TappedRoutedEventArgs is needed to handle Tapped event. 
// This is because Windows Runtime requires class names registered at compile time.
//
require('Windows.UI.Xaml.Input.TappedRoutedEventArgs');

var win = Ti.UI.createWindow({ backgroundColor: 'green' }),
    Button = require('Windows.UI.Xaml.Controls.Button'),
    button = new Button();

button.Content = "PUSH";
button.addEventListener('Tapped', function (e) {
    var pos = e.GetPosition(null);
    Ti.API.info(pos.X + ' -' + pos.Y);
});

win.add(button);
win.open();
```

Note: This requires new hyperloop with native event handling support (which will be pushed later on).


## How to Test

- Download Titanium SDK build artifacts from here: [titanium_mobile_windows_10_prs/1279/artifact/dist/windows](http://studio-jenkins.appcelerator.org/job/titanium_mobile_windows_10_prs/1279/artifact/dist/windows/) as well as [titanium_mobile_windows_8.1_prs/1021/artifact/dist/windows](http://studio-jenkins.appcelerator.org/job/titanium_mobile_windows_8.1_prs/1021/artifact/dist/windows/), and extract/replace them with your Titanium SDK `windows` directory.
- Make sure to installed latest `hyperloop` module and you have Appcelerator Platform Pro account
